src/nix/nix_store.rs

```diff
@@ -84,20 +88,30 @@ impl NixStoreCmd {
     }
 
     /// Return the derivation used to build the given build output.
-    async fn nix_store_query_deriver(&self, out_path: PathBuf) -> Result<DrvOut> {
+    async fn nix_store_query_deriver(&self, out_path: PathBuf) -> Result<DrvOut, NixCmdError> {
         let mut cmd = self.command();
         cmd.args(["--query", "--deriver", &out_path.to_string_lossy().as_ref()]);
         nix_rs::command::trace_cmd(&cmd);
-        let out = cmd.output().await?;
+        let out = cmd
+            .output()
+            .await
+            .map_err(CommandError::ChildProcessError)?;
         if out.status.success() {
             let drv_path = String::from_utf8(out.stdout)?.trim().to_string();
+            if drv_path.contains("unknown-deriver") {
+                let exit_code = Some(1);
+                let stderr =
+                    Some("nix-store --query --deriver returned UnknownDeriver".to_string());
+                return Err(NixCmdError::from(CommandError::ProcessFailed {
+                    stderr,
+                    exit_code,
```

Comment:
We should not "fake" error messages. Here, neither the process failed, nor did it print this particular message in stderr. Just create a new error `struct`, with a constructor called `UnknownDeriver`.

Remember, 
- Explicit is better than implicit.
https://peps.python.org/pep-0020/